Automattic/wp-calypso
# Packages: Add typescript relevant deps to calypso-build

#### Changes proposed in this Pull Request

Packages: Add typescript relevant deps to calypso-build

See https://github.com/Automattic/newspack-blocks/pull/7 :roll_eyes: -- That build currently fails because `@automattic/calypso-build`'s `babel.config.js` [is referencing `'@babel/preset-typescript'`](https://github.com/Automattic/wp-calypso/blob/f2e8360617795326bcf1cc6856d745003decb477/packages/calypso-build/babel.config.js#L13
) even though it's not a dependency of that package (this was introduced by #31902).

I think it makes sense to move all Typescript relevant deps from the top-level `package.json` to `@automattic/calypso-build`'s.

#### Testing instructions

`npm link` from `newspack-blocks`, run `npm run build` there.

Also, verify that Calypso still builds and runs! (`npm start`, and/or CI)

#### Follow-up

As @simison pointed out, it probably makes sense to add some kind of out-of-context build using `@automattic/calypso-build` to CI to make sure it continues to work if we make any changes to it.
